src/activities/bargame/bargame.js

```diff
@@ -0,0 +1,212 @@
+    }
+    currentLevel = level;
+    sublevel = 1;
+
+    reSetup();
+    initLevel();
+}
+
+function calculateWinPlaces() {
+    var winners = [];
+    var winnersList = [];
+    var min = numberBalls[sublevel - 1][0];
+    var max = numberBalls[sublevel - 1][1];
+    var period = (min + max);
+
+    for (var x = 0; x < min; x++) {
+        winnersList.push((boardSize[sublevel - 1] - 1 - x) % period);
+    }
+
+    for (var x = 0; x < (boardSize[sublevel - 1]); x++ ) {
+        if (winnersList.indexOf((x + 1) % period) >= 0) {
+            winners.push(x);
+        }
+    }
+
+    var level_win = (level - 1) * min;
+
+    if (level_win == 0) {
+        winners = [];
+    } else {
+        winners = winners.slice(-level_win);
+    }
+
+    listWin =  winners;
+}
+
+function machinePlay() {
+    function accessible(x) {
+        if (listWin.indexOf(x + moveCount) >= 0) {
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    function randomNumber(minimum, maximum){
+        return Math.round( Math.random() * (maximum - minimum) + minimum);
+    }
+
+    var playable = [];
+
+    for (var x = numberBalls[sublevel - 1][0]; x < numberBalls[sublevel - 1][1] + 1; x++) {
+        if (accessible(x)) {
+            playable.push(x);
+        }
+    }
+    if (playable.length != 0) {
+        var value = playable[Math.floor(Math.random()*playable.length)];
+    } else {
+        var value = randomNumber(numberBalls[sublevel - 1][0], numberBalls[sublevel - 1][1]);
+    }
+
+    play(false, value);
+}
+
+function play(human, value) {
+    for (var x = 0; x < value ; x++) {
+        moveCount++;
+        if (moveCount <= (boardSize[sublevel - 1] - 1)) {
+            if (human == true) {
+                items.answerBallsPlacement.children[moveCount].opacity = 1.0;
+                items.answerBallsPlacement.children[moveCount].source = url + "green_ball.svg";
+            } else {
+                items.answerBallsPlacement.children[moveCount].opacity = 1.0;
+                items.answerBallsPlacement.children[moveCount].source = url + "blue_ball.svg";
+            }
+        }
+        if (moveCount == (boardSize[sublevel - 1] - 1)) {
+            // Sublevel increment
+            sublevel++;
+            if (sublevel > numberOfSublevel) {
+                sublevel = 1;
+                level++;
+                if (level > maxlevel) {
+                    level = maxlevel;
+                    sublevel = 1;
+                }
+                currentLevel = level;
+            }
+            if (human == false) {
+                items.bonus.good("flower");
+            } else {
+                items.bonus.bad("flower");
+            }
+            return;
+        }
+    }
+    if (human == true) {
+        machinePlay();
+    }
+}
+
+// Refreshes the scene
+function reSetup() {
+    items.tuxArea.enabled = true;
+    items.tuxArea.hoverEnabled = true;
+    moveCount = -1;
+    items.numberOfBalls.text = "1";
```

Comment:
No, the minimum number of balls (according to the GTK+ version) at the start of a level should be 1 by default. This would change to the 'minimum number' for that level in every next moves.

It's like giving the kid a first chance to start with a small move to understand tux's strategy.